File: AircraftSimulatorsControl/DTO/API/SimulatorRequest.cs
namespace AircraftSimulatorsControl.DTO.API
{
    public class SimulatorRequest
    {

        public string Model { get; set; } = "Unknown model";

        public string Type { get; set; } = "Unknown type";

        public string SimulatorName { get; set; } = "Unknown name";

        public DateOnly ProductionDate { get; set; }

        public DateOnly CommissioningDate { get; set; }

        public int TechFrequrency { get; set; }

        public List<Component> Components { get; set; } = new List<Component>();

        public class Component
        {

            public string Name { get; set; } = "Unknown component";

        }

    }
}

File: WebServer/src/main/java/org/server/api/dto/simulator/SimulatorRequest.java
package org.server.api.dto.simulator;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SimulatorRequest {

    @JsonProperty
    private String model;

    @JsonProperty
    private String type;

    @JsonProperty
    private String simulatorName;

    @JsonProperty
    private LocalDate productionDate;

    @JsonProperty
    private LocalDate commissioningDate;

    @JsonProperty
    private LocalDate lastTechCheckDate;

    @JsonProperty
    private int techCheckFrequency;

    @JsonProperty
    private List<ComponentRequest> components;

}
